pkg/kubelet/kubelet_pods.go

```diff
@@ -1224,33 +1246,27 @@ func (pk *podKillerWithChannel) KillPod(pair *kubecontainer.PodPair) {
 // PerformPodKillingWork launches a goroutine to kill a pod received from the channel if
 // another goroutine isn't already in action.
 func (pk *podKillerWithChannel) PerformPodKillingWork() {
-	killing := sets.NewString()
-	// guard for the killing set
-	lock := sync.Mutex{}
 	for podPair := range pk.podKillingCh {
 		runningPod := podPair.RunningPod
 		apiPod := podPair.APIPod
 
-		lock.Lock()
-		exists := killing.Has(string(runningPod.ID))
-		if !exists {
-			killing.Insert(string(runningPod.ID))
+		if pk.IsPodPendingTerminationByUID(runningPod.ID) {
```

Comment:
Should we log that it's pending termination here too?

Reply:
I don't want it to be too verbose... I'm on the fence.

Reply:
Update: I had already added one... Should be sufficient.

https://github.com/kubernetes/kubernetes/pull/98424/files#diff-e81aa7518bebe9f4412cb375a9008b3481b19ec3e851d3187b3021ee94148f0dR1239